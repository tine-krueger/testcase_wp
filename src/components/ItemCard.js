import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { getExcerpt } from '../hooks/useItems'

ItemCard.propTypes = {
    item: PropTypes.array
}

export default function ItemCard(item) {
    const excerpt = getExcerpt(item.item[2])
  
    return(
        <Card key={item.index}>
            <img src={item.item[0]} alt='Speicherstadt-Cristina-Gottardi/unsplash'/>
            <Text>
                <h3>{item.item[1]}</h3>
                <p>{excerpt}</p>
            </Text>
        </Card>
    )
}

const Card = styled.li`
    background: var(--cards);
    display: grid;
    gap: 0 1em;
    margin-bottom: 2em;
    
    img {
        max-width: 100%;
        height: auto;
    }

    @media (min-width: 768px) {
        grid-template-columns: 1fr 1fr;
    }

    @media (min-width: 992px) {
        grid-template-columns: 2fr 5fr;
    }

    @media (min-width: 1200px) {
        grid-template-columns: 1fr 3fr;
    }
`
const Text = styled.div`
    padding: .5em 1em;

    h3 {
        margin-top: .5em;
        margin-bottom: .5em;
    }

    p {
        font-size: .8em;
    }

    @media (min-width: 992px) {
        padding: 0 1em;

        p {
            margin: .75em 0;
        }
    }

    @media (min-width: 1200px) {
        p {
            font-size: 1em;
        }
    }
`
