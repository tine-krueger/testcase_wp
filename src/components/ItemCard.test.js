import { render } from '@testing-library/react'
import 'jest-styled-components'
import ItemCard from './ItemCard'

const item = [
        '../../public/img/speicherstadt_600.jpg',
        'Headline 1',
        'Text example'
]

describe ('Item Card', () => {
    it('renders correctly', () => {
        const { container } = render(
            <ItemCard
                item={item}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('shows the correct item information', () => {
        const { getByRole, getByText } = render(
            <ItemCard
                item={item}
            />
        )
        expect(getByRole('img', {name: /Speicherstadt-Cristina-Gottardi/i })).toBeInTheDocument()
        expect(getByRole('heading', {name:/Headline 1/i})).toBeInTheDocument()
        expect(getByText(/Text example/i)).toBeInTheDocument()
    })
})