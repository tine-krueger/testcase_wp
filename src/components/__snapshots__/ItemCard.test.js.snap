// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Card renders correctly 1`] = `
.c0 {
  background: var(--cards);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 2em;
}

.c0 img {
  max-width: 100%;
  object-fit: cover;
}

.c1 {
  padding: .5em 1em;
}

.c1 h3 {
  margin-top: .5em;
  margin-bottom: .5em;
}

.c1 p {
  font-size: .8em;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 img {
    max-width: 45%;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

@media (min-width:992px) {
  .c0 img {
    max-width: 30%;
  }
}

@media (min-width:992px) {
  .c1 {
    padding: 0 1em;
  }

  .c1 p {
    margin: .75em 0;
  }
}

@media (min-width:1200px) {
  .c1 p {
    font-size: 1em;
  }
}

<div>
  <li
    class="c0"
  >
    <img
      alt="Speicherstadt-Cristina-Gottardi/unsplash"
      src="../../public/img/speicherstadt_600.jpg"
    />
    <div
      class="c1"
    >
      <h3>
        Headline 1
      </h3>
      <p>
        Text example
      </p>
    </div>
  </li>
</div>
`;
